app/grandchallenge/components/models.py

```diff
@@ -2062,58 +2088,146 @@ def assign_permissions(self):
+                Notification.send(
+                    kind=notification_type,
+                    actor=user_upload.creator,
+                    message=f"Your file upload failed with the following error: {error_message}",
+                    target=self.base_object,
+                    description=error_message,
+                )
+        elif (
+            notification_type
+            == NotificationType.NotificationTypeChoices.IMAGE_IMPORT_STATUS
+        ):
+            if not upload_session:
+                raise RuntimeError(
+                    "For image import status notifications, "
+                    "you must provide an upload_session"
+                )
+            else:
+                Notification.send(
+                    kind=notification_type,
+                    message=f"failed with the following error: {error_message}",
+                    action_object=upload_session,
+                )
+        else:
+            raise RuntimeError(
+                f"Notification type {notification_type} is not supported."
+            )
+
+
 class CIVForObjectMixin:
-    def create_civ(self, *, ci_slug, new_value, user=None):
+    def create_civ(self, *, ci_slug, new_value, user=None, linked_task=None):
         ci = ComponentInterface.objects.get(slug=ci_slug)
-        try:
-            current_civ = self.values.filter(interface=ci).get()
-        except ObjectDoesNotExist:
-            current_civ = None
-        except MultipleObjectsReturned as e:
-            raise e
+        current_civ = self.get_current_value_for_interface(interface=ci)
 
         if ci.is_json_kind and not ci.requires_file:
             return self.create_civ_for_value(
-                ci=ci, current_civ=current_civ, new_value=new_value
+                ci=ci,
+                current_civ=current_civ,
+                new_value=new_value,
+                linked_task=linked_task,
             )
         elif ci.is_image_kind:
             return self.create_civ_for_image(
-                ci=ci, current_civ=current_civ, new_value=new_value, user=user
+                ci=ci,
+                current_civ=current_civ,
+                new_value=new_value,
+                user=user,
+                linked_task=linked_task,
             )
         elif ci.requires_file:
             return self.create_civ_for_file(
-                ci=ci, current_civ=current_civ, new_value=new_value
+                ci=ci,
+                current_civ=current_civ,
+                new_value=new_value,
+                linked_task=linked_task,
             )
         else:
             NotImplementedError(f"CIV creation for {ci} not handled.")
 
-    def create_civ_for_value(self, *, ci, current_civ, new_value):
+    def create_civ_for_value(self, *, ci, current_civ, new_value, linked_task):
         current_value = current_civ.value if current_civ else None
-        civ = ComponentInterfaceValue(interface=ci, value=new_value)
+
+        if new_value is not None:
```

Comment:
We can have a CIV with value `None` if that passes validation. 

Not sure why I added this if-statement here  though. It seems unnecessary since creating a CIV without passing a value is the same as creating one with value=None. Will remove!

Reply:
Ah, I did that to avoid saving a new CIV to the database when the new value was None, but doesn't matter that much, and it would anyway be reused, so it would at most mean creating 1 CIV with value=None per interface.